From d0627f485b75712880d3532265b6b02115dff8c7 Mon Sep 17 00:00:00 2001
From: Jaime Frey <jfrey@cs.wisc.edu>
Date: Tue, 2 Aug 2016 16:19:54 -0500
Subject: [PATCH 1/4] Fix linux distro detection for rhel7 and fedora. #5822

/etc/issue doesn't have the distro's name on these distros.
---
 build/cmake/macros/SystemSpecificInformations.cmake | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/build/cmake/macros/SystemSpecificInformations.cmake b/build/cmake/macros/SystemSpecificInformations.cmake
index dcf575e..9f2fe29 100644
--- a/build/cmake/macros/SystemSpecificInformations.cmake
+++ b/build/cmake/macros/SystemSpecificInformations.cmake
@@ -106,11 +106,15 @@ endif(WIN32)
 if(UNIX)
   if(CMAKE_SYSTEM_NAME MATCHES "Linux")
     set(SPECIFIC_SYSTEM_VERSION_NAME "${CMAKE_SYSTEM_NAME}")
     if(EXISTS "/etc/issue")
       set(LINUX_NAME "")
-      file(READ "/etc/issue" LINUX_ISSUE)
+      if(EXISTS "/etc/redhat-release")
+        file(READ "/etc/redhat-release" LINUX_ISSUE)
+      else()
+        file(READ "/etc/issue" LINUX_ISSUE)
+      endif()
       # Fedora case
       if(LINUX_ISSUE MATCHES "Fedora")
         string(REGEX MATCH "release ([0-9]+)" FEDORA "${LINUX_ISSUE}")
         set(LINUX_NAME "Fedora")
 	set(LINUX_VER "${CMAKE_MATCH_1}")
-- 
2.6.3

